cmd/broadcaster-cli/app/keyset/address/address.go

```diff
@@ -27,38 +24,14 @@ var (
 			keysFlag := viper.GetString("keys")
 			selectedKeys := strings.Split(keysFlag, ",")
 
-			var keys map[string]string
-			err = viper.UnmarshalKey("privateKeys", &keys)
+			keys, err := helper.GetPrivateKeys()
 			if err != nil {
 				return err
 			}
 
-			var keySets []*keyset.KeySet
-
-			if len(keys) > 0 {
-				if len(selectedKeys) > 0 {
-					for _, selectedKey := range selectedKeys {
-						key, found := keys[selectedKey]
-						if !found {
-							return fmt.Errorf("key not found: %s", selectedKey)
-						}
-						fundingKeySet, _, err := helper.GetKeySetsXpriv(key)
-						if err != nil {
-							return fmt.Errorf("failed to get key sets: %v", err)
-						}
-						keySets = append(keySets, fundingKeySet)
-					}
-				} else {
-					for _, key := range keys {
-						fundingKeySet, _, err := helper.GetKeySetsXpriv(key)
-						if err != nil {
-							return fmt.Errorf("failed to get key sets: %v", err)
-						}
-						keySets = append(keySets, fundingKeySet)
-					}
-				}
-			} else {
-				return errors.New("no keys given in configuration")
+			keySets, err := helper.GetKeySets(keys, selectedKeys)
+			if err != nil {
+				return err
```

Comment:
Just loud thinking: maybe it could be one function too? I mean we could have helper functions like:
`func GetPrivateKeys()`
`func GetKeySetsFor(keys map[string]string (...))`
`func GetKeySets()`

The first two would be for future flexibility, and the last one would be used in the currently existing implementation. What do you think?

Reply:
I have them as getKeySets and GetPrivateKeys , yeah if need be i can fragment and add one more function  GetKeySetsFor(keys map[string]string (...))

Reply:
we also have this function GetKeySetsKeyFile() in the helper which will resolve the issue ?